scripts/popup.js

```diff
@@ -221,20 +221,10 @@ function show_news() {
     var to_check_url = url.replace(/^https?:\/\/(www\.)?/, '')
     var news_host = to_check_url.split('.')[0]
     tabId = tabs[0].id
-    var set_of_sites = new Set([
-      'apnews',
-      'factcheck',
-      'forbes',
-      'huffingtonpost',
-      'nytimes',
-      'politifact',
-      'snopes',
-      'theverge',
-      'usatoday',
-      'vox',
-      'washingtonpost'
-    ])
-    chrome.storage.sync.get(['news', 'show_context'], function (event) {
+
+    chrome.storage.sync.get(['news', 'newshosts', 'show_context'], function (event) {
+      var set_of_sites = new Set(event.newshosts.map(host => host.slice(0, host.indexOf("."))))
```

Comment:
I think that you should move the creation of `set_of_sites` in `scripts/background.js` and save it to `chrome.storage` along with `newshosts`.

This way, the Set creation will happen only once and you'll get the Set ready to use from `chrome.storage`.

You won't get `newshosts` in `scipts/popup.js`, you'll get `set_of_sites`.